include/seqan3/alphabet/structure/dssp9.hpp

```diff
@@ -0,0 +1,339 @@
+        return *this;
+    }
+    //!\}
+
+    //!\brief The size of the alphabet, i.e. the number of different values it can take.
+    static constexpr rank_type value_size{9};
+
+    //!\name Comparison operators
+    //!\{
+    constexpr bool operator==(dssp9 const & rhs) const noexcept
+    {
+        return _value == rhs._value;
+    }
+
+    constexpr bool operator!=(dssp9 const & rhs) const noexcept
+    {
+        return _value != rhs._value;
+    }
+
+    constexpr bool operator<(dssp9 const & rhs) const noexcept
+    {
+        return _value < rhs._value;
+    }
+
+    constexpr bool operator>(dssp9 const & rhs) const noexcept
+    {
+        return _value > rhs._value;
+    }
+
+    constexpr bool operator<=(dssp9 const & rhs) const noexcept
+    {
+        return _value <= rhs._value;
+    }
+
+    constexpr bool operator>=(dssp9 const & rhs) const noexcept
+    {
+        return _value >= rhs._value;
+    }
+    //!\}
+
+protected:
+    //!\privatesection
+    /*!\brief The internal type is a strictly typed enum.
+     *
+     * This is done to prevent aggregate initialization from numbers and/or chars.
+     * It is has the drawback that it also introduces a scope which in turn makes
+     * the static "letter values " members necessary.
+     */
+    enum struct internal_type : rank_type
+    {
+        H, B, E, G, I, T, S, C, X
+    };
+
+    //!\brief Value to char conversion table.
+    static constexpr char_type value_to_char[value_size]
+    {
+        'H', 'B', 'E', 'G', 'I', 'T', 'S', 'C', 'X'
+    };
+
+    //!\brief Char to value conversion table.
+    static constexpr std::array<internal_type, 256> char_to_value
+    {
+    [] () constexpr
+    {
+        using in_t = internal_type;
+        std::array<in_t, 256> ret{};
+
+        // initialize with UNKNOWN (std::array::fill unfortunately not constexpr)
+        for (auto & c : ret)
+            c = in_t::X;
+
+        // canonical
+        ret['H'] = in_t::H;
+        ret['B'] = in_t::B;
+        ret['E'] = in_t::E;
+        ret['G'] = in_t::G;
+        ret['I'] = in_t::I;
+        ret['T'] = in_t::T;
+        ret['S'] = in_t::S;
+        ret['C'] = in_t::C;
+        ret['X'] = in_t::X;
+
+        return ret;
+    } ()
+    };
+
+public:
+    //!\privatesection
+    //!\brief The data member.
+    internal_type _value;
+    //!\publicsection
+};
+
+constexpr dssp9 dssp9::H{internal_type::H};
+constexpr dssp9 dssp9::B{internal_type::B};
+constexpr dssp9 dssp9::E{internal_type::E};
+constexpr dssp9 dssp9::G{internal_type::G};
+constexpr dssp9 dssp9::I{internal_type::I};
+constexpr dssp9 dssp9::T{internal_type::T};
+constexpr dssp9 dssp9::S{internal_type::S};
+constexpr dssp9 dssp9::C{internal_type::C};
+constexpr dssp9 dssp9::X{internal_type::X};
+
+} // namespace seqan3
+
+namespace seqan3::detail
+{
+
+//!\brief seqan3::dssp9 is defined as being a structure alphabet.
+//!\ingroup structure
+template <>
+struct is_structure<dssp9> : public std::true_type
+{};
```

Comment:
like I said above, we should rethink this. It especially doesnt make sense to have a common concept for amino acid and rna structure, since you never really use them interchangeably.